dubbo-serialization-extensions/dubbo-serialization-fury/src/main/java/org/apache/dubbo/common/serialize/fury/dubbo/FuryObjectInput.java

```diff
@@ -40,7 +40,7 @@ public FuryObjectInput(Fury fury, MemoryBuffer buffer, InputStream input) {
 
   @Override
   public Object readObject() {
-    return fury.deserialize(new FuryInputStream(input, buffer.size()));
+    return fury.deserializeJavaObjectAndClass(new FuryInputStream(input, buffer.size()));
```

Comment:
We should not always create a new FuryInputStream here, but instead use the same one. We should create a FuryInputStream when constructing FuryObjectInput

Reply:
If the passed `InputStream` is used by dubbo after fury consumed more data, the usage in dubbo may fail. If we created `FuryInputStream`, we should use `FuryInputStream` in all places when using original `InputStream` just like we use `BufferedInputStream`